stdlib/Serialization/src/Serialization.jl

```diff
@@ -748,6 +748,21 @@ function resolve_ref_immediately(s::AbstractSerializer, @nospecialize(x))
     nothing
 end
 
+function gettable(s::AbstractSerializer, id::Int)
+    if haskey(s.table, id)
+        return s.table[id]
```

Comment:
I would worry about the performance implications of this double lookup --- in the common case the key will be found.

Reply:
Can we use `get(s.table, id, -1)`?

Or perhaps `get(s.table, id, secret_table_token)`?

Reply:
Updated to use this form: 
```
get(s.table, id) do
    errmsg = ...
    error(errmsg)
end
```